src/coordinator/roles.spec.js

```diff
@@ -29,6 +29,200 @@ describe('Coordinator/Roles', function() {
     this.sandbox.restore();
   });
 
+  describe('addApplication', function() {
+    context('when all the required parameters are provided', function() {
+      let expectedRoleApplication;
+      let promise;
+      let request;
+      let application;
+      let organization;
+      let role;
+      let roleApplicationFromServer;
+      let toCamelCase;
+
+      beforeEach(function() {
+        organization = fixture.build('contxtOrganization');
+        application = fixture.build('contxtApplication');
+        role = fixture.build('contxtRole');
+
+        expectedRoleApplication = fixture.build('contxtRoleApplication');
+        roleApplicationFromServer = fixture.build(
+          'contxtRoleApplication',
+          expectedRoleApplication,
+          {
+            fromServer: true
+          }
+        );
+
+        request = {
+          ...baseRequest,
+          post: this.sandbox.stub().resolves(roleApplicationFromServer)
+        };
+        toCamelCase = this.sandbox
+          .stub(objectUtils, 'toCamelCase')
+          .callsFake((app) => expectedRoleApplication);
+
+        const roles = new Roles(baseSdk, request, expectedHost);
+        promise = roles.addApplication(organization.id, role.id, application.id);
+      });
+
+      it('posts the role application to the server', function() {
+        expect(request.post).to.be.calledWith(
+          `${expectedHost}/applications/${application.id}/roles/${role.id}`
+        );
+      });
+
+      it('formats the returned role application', function() {
+        return promise.then(() => {
+          expect(toCamelCase).to.be.calledWith(roleApplicationFromServer);
+        });
+      });
+
+      it('returns a fulfilled promise', function() {
+        return expect(promise).to.be.fulfilled;
+      });
```

Comment:
@leeallen337 perhaps I should just not return the mapping data.  Is it useful?

Reply:
Generally speaking, all our `POST` endpoint return the item created, even if you may just be posting to some type of join table or the data isn't necessarily useful, so I don't know if it would be a good idea to deviate from that pattern. It might be a little inconsistent to have some `POST`s return data while others don't

You could wrap it into one test (I think you can chain it like below) and do something like
```
it('returns a fulfilled promise with the data', function() {
  return expect(promise).to.be.fulfilled
    .and.to.eventually.deep.equal(expectedRoleApplication);
});
```
Maybe that works? What do you think?

Reply:
dones